### DISCUSSION TITLE
ConversationHandler Question

### DISCUSSION THREAD

**Discussion by shapiroj18** [python-telegram-bot/python-telegram-bot#2466] [Q&A]
Hey all, I am trying to get a ConversationHandler working with multiple steps. I have pasted my code here: https://pastebin.com/FZGvtTuq.

The general flow I'm hoping for is:
User types `/randomjam`. The bot then asks if they were hoping for `True Random`, `Year Random`, `Song Random`, or `YearSong Random`. Depending on their response, it will have them specify additional data needed (year or song via a message). Then, I'm hoping to process a different function dependent on which type of random they selected.

Right now, the handler can make it to step two (asking the user for input dependent on what type of random they select), but never moves to step three (processing function and sending response data based on their input data).

Any help would be appreciated!

**Comment by Bibo-Joshi** [comment]
The problem is that for your `*_RANDOM` states you're filtering for the wrong input: The filters say that you expect the input of the keyboard you attached in `random_job`, but that's already handled in the `RANDOMJAM` state. E.g. in `YEAR_RANDOM` you should be expecting a year as input from the user instead of the word "Year".
